### PR TITLE
Add support for maps

### PR DESCRIPTION
**Pull request by paddycarver**

### CODE REVIEW

**Comment by paddycarver** [comment]
Depends on #35.

File: types/map.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/internal/reflect"
+
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+// MapType is an AttributeType representing a list of values. All values must
+// be of the same type, which the provider must specify as the ElemType
+// property. Keys will always be strings.
```
Comment (bschaeffer) [inline types/map.go:15]: Copy pasta?

Comment (bschaeffer) [inline types/map.go:15]: Copy paste error?

**Comment by paddycarver** [comment]
Looks good to me. Mind putting a `.changelog/37.txt` file in there, @kmoe, and then it's good to merge?

**Review by kmoe — APPROVED** [review]

**Pull request merged**